src/bindings/python/tests/test_frontend/test_frontend_onnx.py

```diff
@@ -593,3 +608,47 @@ def test_op_extension_via_frontend_extension_map_attributes():
 
     model = ie.read_model(onnx_model_for_op_extension_test)
     assert model
+
+
+def get_builtin_extensions_path():
+    import openvino.frontend
+    python_frontend_path = openvino.frontend.__path__[0]
+    lib_folder_pos = python_frontend_path.rfind("lib/")
+    if lib_folder_pos != -1:
+        lib_folder_path = python_frontend_path[:lib_folder_pos + 4]
+        for file in os.listdir(lib_folder_path):
+            if file == "libtest_builtin_extensions_1.so" or file == "libtest_builtin_extensions_1.dll":
+                return os.path.join(lib_folder_path, file)
+    return ""
+
+
+@pytest.mark.skipif(len(get_builtin_extensions_path()) == 0,
+                    reason="The extension library path was not found")
+def test_so_extension_via_frontend_convert_input_model():
+    skip_if_onnx_frontend_is_disabled()
+
+    def load_model():
+        fe = fem.load_by_framework(framework=ONNX_FRONTEND_NAME)
+        fe.add_extension(get_builtin_extensions_path())
+        in_model = fe.load(onnx_model_2_filename)
+        return fe.convert(in_model)
+
+    model = load_model()  # model has longer lifetime than frontend
+
+    assert any(op.get_type_name() == "Swish" for op in model.get_ops())
+    assert all(op.get_type_name() != "Relu" for op in model.get_ops())
+
+
+@pytest.mark.skipif(len(get_builtin_extensions_path()) == 0,
+                    reason="The extension library path was not found")
+def test_so_extension_via_frontend_decode_input_model():
+    skip_if_onnx_frontend_is_disabled()
+
+    def load_decoded_model():
+        fe = fem.load_by_framework(framework=ONNX_FRONTEND_NAME)
+        fe.add_extension(get_builtin_extensions_path())
+        in_model = fe.load(onnx_model_2_filename)
+        return fe.decode(in_model)
+
+    decoded_model = load_decoded_model()  # decoded model has longer lifetime than frontend
```

Comment:
Will same C++ test cause the same crash? I.e. when user loads frontend, adds extension, converts to model, and then holds only `ov::Model` object? My suggestion is to consider storing list of loaded extensions into `[m_shared_object](https://github.com/openvinotoolkit/openvino/blob/5724c5ac44e912ae1cd11543fae00526be436ba8/src/frontends/common/include/openvino/frontend/frontend.hpp#L25)` class member so that created ov::Model will also hold a shared pointer to it on creation